bin/ncp-diag

```diff
@@ -66,78 +66,52 @@ echo "Internet check|$( ping -W 2 -w 1 -q github.com &>/dev/null && echo ok || e
 
 function is_port_open()
 {
-  local port_url tmp_file token ipv4_portcheck_args ipv6_portcheck_args
-  tmp_file=$(mktemp)
-  trap 'rm -rf ${tmp_file}' EXIT ERR SIGINT SIGQUIT SIGABRT SIGTERM SIGHUP
-
-  local port="${1?}"
-  local publicIPv4="${2}"
-  local publicIPv6="${3}"
-
-  readonly port_url="https://portchecker.co"
-
-  if [[ -z "$publicIPv4" ]] && [[ -z "$publicIPv6" ]]
-  then
-    echo -n "Error - IPv4 & IPv6: [N/A] Couldn't get public IP."
-    return 1
-  fi
-
-  token=$(wget -T2 -t1 -qO- --keep-session-cookies --save-cookies "${tmp_file}" "${port_url}" | grep -oP "_csrf\" value=\"\K.*\"" )
-  readonly ipv4_portcheck_args=(-T2 -t1 -qO- --load-cookies "${tmp_file}" "${port_url}/check" --post-data "target_ip=${publicIPv4}&port=${port}&_csrf=${token::-1}")
-  readonly ipv6_portcheck_args=(-T2 -t1 -qO- --load-cookies "${tmp_file}" "${port_url}/check" --post-data "target_ip=${publicIPv6}&port=${port}&_csrf=${token::-1}")
-
-  [[ -n "${token}" ]] || {
-    echo -n "Error - Couldn't obtain a token for port check"
-    return 1
-  }
-
-  local ipv4_port_access=False
-  local ipv6_port_access=False
-  [[ -n "$publicIPv4" ]] && \
-    grep -q '<span class="green">open</span>' <(wget "${ipv4_portcheck_args[@]}") && \
-    ipv4_port_access=True
-  [[ -n "$publicIPv6" ]] && \
-    grep -q '<span class="green">open</span>' <(wget "${ipv6_portcheck_args[@]}") && \
-    ipv6_port_access=True
-
-  local result=""
-  if [[ "${ipv4_port_access}" == True ]] || [[ "${ipv6_port_access}" == True ]]
-  then
-    result="open ("
-  else
-    result="closed"
-  fi
-
-  [[ "${ipv4_port_access}" == True ]] && result="${result}ipv4)"
-
-  [[ "${ipv6_port_access}" == True ]] && result="${result/)/ \& }ipv6)"
+  # The URL leads to an application I've deployed for NCP on https://fly.io using a Docker container I made.
+  # The image for the container is available on Docker Hub (zendai/checkport:sanic) if you wish to deploy one yourself.
+  # The code for the Sanic server and Docker image is available at: https://github.com/ZendaiOwl/Build/tree/master/Docker/Python/Sanic/checkport
+  # I only have a free tier with limited outbound data per month, 100GB p/month.
+  # If we go over 100GB outbound data in a month, I will start being charged for the data going over that limit.
+  # I used a low level Python socket library & fortunately each request only consumes aprox. ~ 60-74 bytes p/second.
+  # Meaning 100GB should be plenty, it should be enough to handle a little less
+  # than 450 request p/second a month, unless my calculations are wrong.
+  # Thank you :pray: from Victor-ray, S. https://github.com/ZendaiOwl
+  local -r PORTURL="https://checkport.zendai.net.eu.org/check"
```

Comment:
Does this validate the requesting IP? Otherwise your service might be abused by malicious actors to scan for vulnerable servers...

Reply:
The endpoint is a GET only endpoint and checks the ports of the connecting client, it doesn't receive input from anything else, the client ip is fetched from a request header that's handled by the fly.io edge network. 

For validation, it does validate that the client ip is a valid one of ipv4 or ipv6, the ports it checks is hard coded to 80 & 443 together or separately, so there's no way to make it check anything other than the connecting client ip unless spoofing the request. 

However I don't see the point for someone to abuse this maliciously because it's really only useful for someone to check their own port forwarding, cuz if someone is able to spoof their request, they could just check the ports they want directly on whatever host they were scanning with a much larger range instead of going through this endpoint that only checks 80 & 443 

You can see the code for the server here ↓

https://github.com/ZendaiOwl/Build/blob/master/Docker/Python/Sanic/checkport/server.py